app/controllers/new_threads_controller.rb

```diff
@@ -53,10 +54,12 @@ def create
 
   def edit
     @new_thread = NewThread.friendly.find(params[:id])
+    authorize! :edit, @new_thread
   end
```

Comment:
You don't need to add individual `authorize! :edit, @new_thread` in each action if you have already added `authorize_resource` at top of class. They are doing the same thing. Please read these wiki: [Controller Authorization Example](https://github.com/CanCanCommunity/cancancan/wiki/Controller-Authorization-Example) and [Authorizing controller actions](https://github.com/CanCanCommunity/cancancan/wiki/Authorizing-controller-actions)